rfd/0011-database-access.md

```diff
@@ -0,0 +1,661 @@
+to the database:
+
+```
+Connection information for "postgres" has been saved to ~/.pg_service.conf.
+You can connect to the database using the following command:
+
+  $ psql "service=postgres user=<user> dbname=<dbname>"
+
+Or configure environment variables and use regular CLI flags:
+
+  $ eval $(tsh db env)
+  $ psql -U <user> <database>
+```
+
+After successful login, users can use any database user to connect to the
+database, as long as it is allowed by the RBAC rules. For example:
+
+```sh
+$ tsh db login postgres-prod && eval $(tsh db env)
+$ psql -U alice mydb     # allowed by role so will connect
+$ psql -U bob mydb       # allowed by role so will connect
+$ psql -U charlie mydb   # not allowed by role so will deny access
+```
+
+### Connecting to PostgreSQL / service file
+
+After logging into a PostgreSQL database, `tsh` configures an appropriate
+section in the [connection service file](https://www.postgresql.org/docs/9.1/libpq-pgservice.html)
+which is located at `~/.pg_service.conf`.
+
+The service file has ini format and can contain multiple sections where each
+section defines connection parameters for a particular "service". PostgreSQL
+clients can refer to a particular service using `service` connection string
+parameter which all libpq-based clients should recognize (e.g. `psql` and
+`pgAdmin` do).
+
+The section is named after the name of the Teleport database service instance
+a user has logged into with `tsh db login <name>` command. If service file is
+already present on the user's machine, its contents will be preserved as just
+a new section will be configured in it. If the file already happens to have
+a section with the same name, it will get overwritten.
+
+An added section may look like this:
+
+```ini
+[postgres-prod]
+host=proxy.teleport.example.com
+port=3080
+sslmode=verify-full
+sslrootcert=/home/user/.tsh/keys/proxy.teleport.example.com/certs.pem
+sslcert=/home/user/.tsh/keys/proxy.teleport.example.com/alice-x509.pem
+sslkey=/home/user/.tsh/keys/proxy.teleport.example.com/alice
+```
+
+Users can then connect to the database using the following command:
+
+```sh
+$ psql "service=postgres-prod user=alice dbname=metrics"
+```
+
+Alternatively, `tsh` can output a set of environment variables supported by
+PostgreSQL clients for users to set in their session:
+
+```sh
+$ tsh db env
+export PGHOST=proxy.teleport.example.com
+export PGPORT=3080
+export PGSSLMODE=verify-full
+export PGSSLROOTCERT=/home/user/.tsh/keys/proxy.teleport.example.com/certs.pem
+export PGSSLCERT=/home/user/.tsh/keys/proxy.teleport.example.com/alice-x509.pem
+export PGSSLKEY=/home/user/.tsh/keys/proxy.teleport.example.com/alice
+$ eval $(tsh db env)
+$ psql -U alice metrics
+```
+
+## Audit events
+
+The following new events are emitted to the Teleport audit log for database
+sessions.
+
+* `db.session.start` fires upon successful connection to the database.
+
+```json
+{
+  "code": "TDB00I",
+  "db_database": "postgres",
```

Comment:
* `db_name` is the name of the entire database instance (DBMS) that's being proxied by Teleport, i.e. the one specified in the config file.
* `db_database` is the name of a single database within that DBMS.

I realize this can be confusing, I open for better suggestions. Maybe rename db_name to db_service?

Reply:
`db_service` is also perhaps a little confusing because it's the terminology Postgres uses to identify a connection method, right? Maybe `teleport_db_service` would be clearer.

Reply:
Ended up renaming to just `service`, similar to how other Teleport-specific fields have no prefix (like, `user`).